FirebaseDatabase/Sources/Api/FIRDatabase.m

```diff
@@ -153,13 +153,32 @@ - (FIRDatabaseReference *)referenceFromURL:(NSString *)databaseUrl {
     }
     FParsedUrl *parsedUrl = [FUtilities parseUrl:databaseUrl];
     [FValidation validateFrom:@"referenceFromURL:" validURL:parsedUrl];
-    if (![parsedUrl.repoInfo.host isEqualToString:_repoInfo.host]) {
-        [NSException
-             raise:@"InvalidDatabaseURL"
-            format:
-                @"Invalid URL (%@) passed to getReference(). URL was expected "
-                 "to match configured Database URL: %@",
-                databaseUrl, [self reference].URL];
+
+    // If in an emulated setting, only validate the host if the underlying
+    // URL isn't a custom host. This allows users to create references from
+    // production URLs in an emulated setting.
+    if (self.repoInfo.underlyingHost != nil) {
```

Comment:
What are your thoughts on matching the newest version of the Android implementation (the one that was merged)? This would allow you to get rid of `underlyingHost` and we would just validate hosts if `_repoInfo.isCustomHost` is false.

Reply:
Would `internalHost` be suitable for validation purposes? I added `underlyingHost` because `internalHost` is mutable and I wasn't sure if it would be appropriate for validating reference URLs.

Reply:
s/Android/JS (sorry I thought this was done on Android already, but looks like it wasn't yet ported back from JS).

See here: https://osscs.corp.google.com/firebase-sdk/firebase-js-sdk/+/master:packages/database/src/api/Database.ts;l=160

This removes the need to check the underlying host and reduces error conditions.

Reply:
Wouldn't this change the behavior by skipping validation for non-emulated environments?

Reply:
It should be the other way around. If we are using an emulated URL, we don't do any validation.

Reply:
Copied the ts implementation.